From 692fe2c33da46aaafb045fef7e7a77ddcbd9cbc3 Mon Sep 17 00:00:00 2001
From: Todor Minchev <todor@minchev.co.uk>
Date: Mon, 16 Feb 2015 19:26:16 +0000
Subject: [PATCH 14/16] .config: add Galileo Gen 2 support

change  LOCALVERSION to 'galileo'
enable x86 Low Power Subsystem Support
enable x86 local APIC support
enable Quark X1000 SPI controller
enable PWM PCA9685 controller

Signed-off-by: Todor Minchev <todor@minchev.co.uk>
---
 .config | 104 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++-------
 1 file changed, 94 insertions(+), 10 deletions(-)

diff --git a/.config b/.config
index a80a302..70680d9 100644
--- a/.config
+++ b/.config
@@ -48,7 +48,7 @@ CONFIG_BROKEN_ON_SMP=y
 CONFIG_INIT_ENV_ARG_LIMIT=32
 CONFIG_CROSS_COMPILE=""
 # CONFIG_COMPILE_TEST is not set
-CONFIG_LOCALVERSION="-galileo-g1"
+CONFIG_LOCALVERSION="-galileo"
 # CONFIG_LOCALVERSION_AUTO is not set
 CONFIG_HAVE_KERNEL_GZIP=y
 CONFIG_HAVE_KERNEL_BZIP2=y
@@ -224,6 +224,7 @@ CONFIG_HAVE_DMA_ATTRS=y
 CONFIG_HAVE_DMA_CONTIGUOUS=y
 CONFIG_GENERIC_SMP_IDLE_THREAD=y
 CONFIG_HAVE_REGS_AND_STACK_ACCESS_API=y
+CONFIG_HAVE_CLK=y
 CONFIG_HAVE_DMA_API_DEBUG=y
 CONFIG_HAVE_HW_BREAKPOINT=y
 CONFIG_HAVE_MIXED_BREAKPOINTS_REGS=y
@@ -327,8 +328,9 @@ CONFIG_X86_FEATURE_NAMES=y
 CONFIG_X86_MPPARSE=y
 CONFIG_X86_EXTENDED_PLATFORM=y
 # CONFIG_X86_GOLDFISH is not set
+CONFIG_X86_INTEL_CE=y
 CONFIG_INTEL_QUARK_X1000_SOC=y
-# CONFIG_X86_INTEL_LPSS is not set
+CONFIG_X86_INTEL_LPSS=y
 # CONFIG_IOSF_MBI is not set
 # CONFIG_X86_RDC321X is not set
 CONFIG_X86_SUPPORTS_MEMORY_FAILURE=y
@@ -387,9 +389,10 @@ CONFIG_PREEMPT_VOLUNTARY=y
 # CONFIG_PREEMPT is not set
 CONFIG_X86_UP_APIC=y
 CONFIG_X86_UP_APIC_MSI=y
-# CONFIG_X86_UP_IOAPIC is not set
+CONFIG_X86_UP_IOAPIC=y
 CONFIG_X86_LOCAL_APIC=y
-# CONFIG_X86_IO_APIC is not set
+CONFIG_X86_IO_APIC=y
+# CONFIG_X86_REROUTE_FOR_BROKEN_BOOT_IRQS is not set
 CONFIG_X86_MCE=y
 CONFIG_X86_MCE_INTEL=y
 # CONFIG_X86_MCE_AMD is not set
@@ -499,6 +502,7 @@ CONFIG_PM_DEBUG=y
 CONFIG_PM_SLEEP_DEBUG=y
 CONFIG_PM_TRACE=y
 CONFIG_PM_TRACE_RTC=y
+CONFIG_PM_CLK=y
 # CONFIG_WQ_POWER_EFFICIENT_DEFAULT is not set
 CONFIG_ACPI=y
 CONFIG_ACPI_LEGACY_TABLES_LOOKUP=y
@@ -553,11 +557,9 @@ CONFIG_CPU_IDLE_GOV_MENU=y
 CONFIG_PCI=y
 # CONFIG_PCI_GOBIOS is not set
 # CONFIG_PCI_GOMMCONFIG is not set
-# CONFIG_PCI_GODIRECT is not set
-CONFIG_PCI_GOANY=y
-CONFIG_PCI_BIOS=y
+CONFIG_PCI_GODIRECT=y
+# CONFIG_PCI_GOANY is not set
 CONFIG_PCI_DIRECT=y
-CONFIG_PCI_MMCONFIG=y
 CONFIG_PCI_DOMAINS=y
 # CONFIG_PCI_CNB20LE_QUIRK is not set
 CONFIG_PCIEPORTBUS=y
@@ -993,6 +995,8 @@ CONFIG_DEBUG_DEVRES=y
 # CONFIG_SYS_HYPERVISOR is not set
 # CONFIG_GENERIC_CPU_DEVICES is not set
 CONFIG_GENERIC_CPU_AUTOPROBE=y
+CONFIG_REGMAP=y
+CONFIG_REGMAP_I2C=m
 CONFIG_DMA_SHARED_BUFFER=y
 # CONFIG_FENCE_TRACE is not set
 
@@ -1004,6 +1008,7 @@ CONFIG_MTD=y
 # CONFIG_MTD_TESTS is not set
 # CONFIG_MTD_REDBOOT_PARTS is not set
 # CONFIG_MTD_CMDLINE_PARTS is not set
+CONFIG_MTD_OF_PARTS=y
 # CONFIG_MTD_AR7_PARTS is not set
 
 #
@@ -1070,6 +1075,23 @@ CONFIG_MTD_CFI_I2=y
 # CONFIG_MTD_LPDDR is not set
 # CONFIG_MTD_SPI_NOR is not set
 # CONFIG_MTD_UBI is not set
+CONFIG_DTC=y
+CONFIG_OF=y
+
+#
+# Device Tree and Open Firmware support
+#
+# CONFIG_OF_UNITTEST is not set
+CONFIG_OF_FLATTREE=y
+CONFIG_OF_EARLY_FLATTREE=y
+CONFIG_OF_ADDRESS=y
+CONFIG_OF_ADDRESS_PCI=y
+CONFIG_OF_IRQ=y
+CONFIG_OF_NET=y
+CONFIG_OF_MDIO=y
+CONFIG_OF_PCI=y
+CONFIG_OF_PCI_IRQ=y
+CONFIG_OF_MTD=y
 CONFIG_ARCH_MIGHT_HAVE_PC_PARPORT=y
 # CONFIG_PARPORT is not set
 CONFIG_PNP=y
@@ -1266,6 +1288,7 @@ CONFIG_NET_VENDOR_AGERE=y
 # CONFIG_NET_VENDOR_AMD is not set
 # CONFIG_NET_XGENE is not set
 CONFIG_NET_VENDOR_ARC=y
+# CONFIG_ARC_EMAC is not set
 # CONFIG_NET_VENDOR_ATHEROS is not set
 # CONFIG_NET_VENDOR_BROADCOM is not set
 # CONFIG_NET_VENDOR_BROCADE is not set
@@ -1306,6 +1329,7 @@ CONFIG_E1000=m
 # CONFIG_NET_PACKET_ENGINE is not set
 # CONFIG_NET_VENDOR_QLOGIC is not set
 CONFIG_NET_VENDOR_QUALCOMM=y
+# CONFIG_QCA7000 is not set
 # CONFIG_NET_VENDOR_REALTEK is not set
 # CONFIG_NET_VENDOR_RDC is not set
 # CONFIG_NET_VENDOR_ROCKER is not set
@@ -1335,6 +1359,7 @@ CONFIG_PHYLIB=y
 #
 # CONFIG_AT803X_PHY is not set
 # CONFIG_AMD_PHY is not set
+# CONFIG_AMD_XGBE_PHY is not set
 # CONFIG_MARVELL_PHY is not set
 # CONFIG_DAVICOM_PHY is not set
 # CONFIG_QSEMI_PHY is not set
@@ -1353,6 +1378,8 @@ CONFIG_PHYLIB=y
 # CONFIG_MICREL_PHY is not set
 # CONFIG_FIXED_PHY is not set
 # CONFIG_MDIO_BITBANG is not set
+# CONFIG_MDIO_BUS_MUX_GPIO is not set
+# CONFIG_MDIO_BUS_MUX_MMIOREG is not set
 # CONFIG_MDIO_BCM_UNIMAC is not set
 # CONFIG_MICREL_KS8995MA is not set
 CONFIG_PPP=m
@@ -1628,6 +1655,7 @@ CONFIG_SERIAL_8250_RSA=y
 CONFIG_SERIAL_CORE=y
 CONFIG_SERIAL_CORE_CONSOLE=y
 # CONFIG_SERIAL_JSM is not set
+# CONFIG_SERIAL_OF_PLATFORM is not set
 # CONFIG_SERIAL_SCCNXP is not set
 # CONFIG_SERIAL_SC16IS7XX is not set
 # CONFIG_SERIAL_TIMBERDALE is not set
@@ -1635,6 +1663,7 @@ CONFIG_SERIAL_CORE_CONSOLE=y
 # CONFIG_SERIAL_ALTERA_UART is not set
 # CONFIG_SERIAL_IFX6X60 is not set
 # CONFIG_SERIAL_PCH_UART is not set
+# CONFIG_SERIAL_XILINX_PS_UART is not set
 # CONFIG_SERIAL_ARC is not set
 # CONFIG_SERIAL_RP2 is not set
 # CONFIG_SERIAL_FSL_LPUART is not set
@@ -1700,12 +1729,15 @@ CONFIG_I2C_HELPER_AUTO=y
 # I2C system bus drivers (mostly embedded / system-on-chip)
 #
 # CONFIG_I2C_CBUS_GPIO is not set
+# CONFIG_I2C_DESIGNWARE_PLATFORM is not set
 # CONFIG_I2C_DESIGNWARE_PCI is not set
 # CONFIG_I2C_EG20T is not set
 # CONFIG_I2C_GPIO is not set
 # CONFIG_I2C_OCORES is not set
 # CONFIG_I2C_PCA_PLATFORM is not set
+# CONFIG_I2C_PXA is not set
 # CONFIG_I2C_PXA_PCI is not set
+# CONFIG_I2C_RK3X is not set
 # CONFIG_I2C_SIMTEC is not set
 # CONFIG_I2C_XILINX is not set
 
@@ -1737,10 +1769,11 @@ CONFIG_SPI_MASTER=y
 # CONFIG_SPI_ALTERA is not set
 CONFIG_SPI_BITBANG=y
 CONFIG_SPI_GPIO=y
+# CONFIG_SPI_FSL_SPI is not set
 # CONFIG_SPI_OC_TINY is not set
 CONFIG_SPI_PXA2XX_DMA=y
 CONFIG_SPI_PXA2XX=m
-# CONFIG_SPI_PXA2XX_PCI is not set
+CONFIG_SPI_PXA2XX_PCI=m
 # CONFIG_SPI_SC18IS602 is not set
 # CONFIG_SPI_TOPCLIFF_PCH is not set
 # CONFIG_SPI_XCOMM is not set
@@ -1781,10 +1814,20 @@ CONFIG_PTP_1588_CLOCK=m
 # Enable PHYLIB and NETWORK_PHY_TIMESTAMPING to see the additional clocks.
 #
 CONFIG_PTP_1588_CLOCK_PCH=m
+CONFIG_PINCTRL=y
+
+#
+# Pin controllers
+#
+# CONFIG_DEBUG_PINCTRL is not set
+# CONFIG_PINCTRL_SINGLE is not set
+# CONFIG_PINCTRL_BAYTRAIL is not set
+# CONFIG_PINCTRL_CHERRYVIEW is not set
 CONFIG_ARCH_WANT_OPTIONAL_GPIOLIB=y
 CONFIG_ARCH_REQUIRE_GPIOLIB=y
 CONFIG_GPIOLIB=y
 CONFIG_GPIO_DEVRES=y
+CONFIG_OF_GPIO=y
 CONFIG_GPIO_ACPI=y
 # CONFIG_DEBUG_GPIO is not set
 CONFIG_GPIO_SYSFS=y
@@ -1792,6 +1835,7 @@ CONFIG_GPIO_SYSFS=y
 #
 # Memory mapped GPIO drivers:
 #
+# CONFIG_GPIO_74XX_MMIO is not set
 # CONFIG_GPIO_GENERIC_PLATFORM is not set
 # CONFIG_GPIO_IT8761E is not set
 # CONFIG_GPIO_F7188X is not set
@@ -1800,6 +1844,7 @@ CONFIG_GPIO_SCH=m
 # CONFIG_GPIO_ICH is not set
 # CONFIG_GPIO_VX855 is not set
 # CONFIG_GPIO_LYNXPOINT is not set
+# CONFIG_GPIO_GRGPIO is not set
 
 #
 # I2C GPIO expanders:
@@ -1810,6 +1855,7 @@ CONFIG_GPIO_PCA953X=m
 # CONFIG_GPIO_PCF857X is not set
 # CONFIG_GPIO_SX150X is not set
 # CONFIG_GPIO_ADP5588 is not set
+# CONFIG_GPIO_ADNP is not set
 
 #
 # PCI GPIO expanders:
@@ -1819,6 +1865,7 @@ CONFIG_GPIO_PCA953X=m
 # CONFIG_GPIO_INTEL_MID is not set
 # CONFIG_GPIO_PCH is not set
 # CONFIG_GPIO_ML_IOH is not set
+# CONFIG_GPIO_SODAVILLE is not set
 # CONFIG_GPIO_RDC321X is not set
 
 #
@@ -1827,6 +1874,7 @@ CONFIG_GPIO_PCA953X=m
 # CONFIG_GPIO_MAX7301 is not set
 # CONFIG_GPIO_MCP23S08 is not set
 # CONFIG_GPIO_MC33880 is not set
+# CONFIG_GPIO_74X164 is not set
 
 #
 # AC97 GPIO expanders:
@@ -1867,6 +1915,7 @@ CONFIG_POWER_SUPPLY=y
 # CONFIG_POWER_AVS is not set
 # CONFIG_HWMON is not set
 CONFIG_THERMAL=y
+CONFIG_THERMAL_OF=y
 CONFIG_THERMAL_DEFAULT_GOV_STEP_WISE=y
 # CONFIG_THERMAL_DEFAULT_GOV_FAIR_SHARE is not set
 # CONFIG_THERMAL_DEFAULT_GOV_USER_SPACE is not set
@@ -1921,8 +1970,10 @@ CONFIG_BCMA_HOST_PCI=y
 CONFIG_MFD_CORE=y
 # CONFIG_MFD_CS5535 is not set
 # CONFIG_MFD_AS3711 is not set
+# CONFIG_MFD_AS3722 is not set
 # CONFIG_PMIC_ADP5520 is not set
 # CONFIG_MFD_AAT2870_CORE is not set
+# CONFIG_MFD_ATMEL_HLCDC is not set
 # CONFIG_MFD_BCM590XX is not set
 # CONFIG_MFD_AXP20X is not set
 # CONFIG_MFD_CROS_EC is not set
@@ -1934,6 +1985,7 @@ CONFIG_MFD_CORE=y
 # CONFIG_MFD_DLN2 is not set
 # CONFIG_MFD_MC13XXX_SPI is not set
 # CONFIG_MFD_MC13XXX_I2C is not set
+# CONFIG_MFD_HI6421_PMIC is not set
 # CONFIG_HTC_PASIC3 is not set
 # CONFIG_HTC_I2CPLD is not set
 # CONFIG_LPC_ICH is not set
@@ -1962,12 +2014,14 @@ CONFIG_LPC_SCH=y
 # CONFIG_MFD_RTSX_PCI is not set
 # CONFIG_MFD_RTSX_USB is not set
 # CONFIG_MFD_RC5T583 is not set
+# CONFIG_MFD_RK808 is not set
 # CONFIG_MFD_RN5T618 is not set
 # CONFIG_MFD_SEC_CORE is not set
 # CONFIG_MFD_SI476X_CORE is not set
 # CONFIG_MFD_SM501 is not set
 # CONFIG_MFD_SMSC is not set
 # CONFIG_ABX500_CORE is not set
+# CONFIG_MFD_STMPE is not set
 # CONFIG_MFD_SYSCON is not set
 # CONFIG_MFD_TI_AM335X_TSCADC is not set
 # CONFIG_MFD_LP3943 is not set
@@ -2652,6 +2706,7 @@ CONFIG_USB_GADGET_STORAGE_NUM_BUFFERS=2
 # CONFIG_USB_NET2280 is not set
 # CONFIG_USB_GOKU is not set
 CONFIG_USB_EG20T=m
+# CONFIG_USB_GADGET_XILINX is not set
 # CONFIG_USB_DUMMY_HCD is not set
 CONFIG_USB_LIBCOMPOSITE=m
 CONFIG_USB_F_ACM=m
@@ -2707,6 +2762,9 @@ CONFIG_MMC_SDHCI_PCI=y
 # CONFIG_MMC_RICOH_MMC is not set
 # CONFIG_MMC_SDHCI_ACPI is not set
 CONFIG_MMC_SDHCI_PLTFM=y
+# CONFIG_MMC_SDHCI_OF_ARASAN is not set
+# CONFIG_MMC_SDHCI_PXAV3 is not set
+# CONFIG_MMC_SDHCI_PXAV2 is not set
 # CONFIG_MMC_WBSD is not set
 # CONFIG_MMC_TIFM_SD is not set
 # CONFIG_MMC_CB710 is not set
@@ -2791,6 +2849,7 @@ CONFIG_RTC_INTF_DEV=y
 # CONFIG_RTC_DRV_DS1374 is not set
 # CONFIG_RTC_DRV_DS1672 is not set
 # CONFIG_RTC_DRV_DS3232 is not set
+# CONFIG_RTC_DRV_HYM8563 is not set
 # CONFIG_RTC_DRV_MAX6900 is not set
 # CONFIG_RTC_DRV_RS5C372 is not set
 # CONFIG_RTC_DRV_ISL1208 is not set
@@ -2849,6 +2908,7 @@ CONFIG_RTC_DRV_CMOS=y
 #
 # on-CPU RTC drivers
 #
+# CONFIG_RTC_DRV_SNVS is not set
 # CONFIG_RTC_DRV_XGENE is not set
 
 #
@@ -2867,8 +2927,10 @@ CONFIG_INTEL_MID_DMAC=m
 # CONFIG_DW_DMAC is not set
 # CONFIG_DW_DMAC_PCI is not set
 # CONFIG_PCH_DMA is not set
+# CONFIG_FSL_EDMA is not set
 CONFIG_DMA_ENGINE=y
 CONFIG_DMA_ACPI=y
+CONFIG_DMA_OF=y
 
 #
 # DMA Clients
@@ -3020,6 +3082,7 @@ CONFIG_IIO_HRTIMER_TRIGGER=m
 # CONFIG_DGNC is not set
 # CONFIG_DGAP is not set
 # CONFIG_GS_FPGABOOT is not set
+# CONFIG_COMMON_CLK_XLNX_CLKWZRD is not set
 CONFIG_X86_PLATFORM_DEVICES=y
 # CONFIG_ACERHDF is not set
 # CONFIG_ASUS_LAPTOP is not set
@@ -3046,6 +3109,17 @@ CONFIG_INTEL_QRK_AUDIO_CTRL=m
 # CONFIG_INTEL_SMARTCONNECT is not set
 # CONFIG_PVPANIC is not set
 # CONFIG_CHROME_PLATFORMS is not set
+CONFIG_CLKDEV_LOOKUP=y
+CONFIG_HAVE_CLK_PREPARE=y
+CONFIG_COMMON_CLK=y
+
+#
+# Common Clock Framework
+#
+# CONFIG_COMMON_CLK_SI5351 is not set
+# CONFIG_COMMON_CLK_SI570 is not set
+# CONFIG_COMMON_CLK_PXA is not set
+# CONFIG_COMMON_CLK_QCOM is not set
 
 #
 # Hardware Spinlock drivers
@@ -3118,6 +3192,7 @@ CONFIG_AD7298=m
 # CONFIG_NAU7802 is not set
 # CONFIG_TI_ADC081C is not set
 # CONFIG_TI_ADC128S052 is not set
+# CONFIG_VF610_ADC is not set
 
 #
 # Amplifiers
@@ -3145,6 +3220,7 @@ CONFIG_AD7298=m
 # CONFIG_AD5791 is not set
 # CONFIG_AD7303 is not set
 # CONFIG_MAX517 is not set
+# CONFIG_MAX5821 is not set
 # CONFIG_MCP4725 is not set
 # CONFIG_MCP4922 is not set
 
@@ -3246,7 +3322,12 @@ CONFIG_IIO_SYSFS_TRIGGER=m
 # CONFIG_VME_BUS is not set
 CONFIG_PWM=y
 CONFIG_PWM_SYSFS=y
-# CONFIG_PWM_LPSS is not set
+# CONFIG_PWM_FSL_FTM is not set
+CONFIG_PWM_LPSS=y
+# CONFIG_PWM_LPSS_PCI is not set
+# CONFIG_PWM_LPSS_PLATFORM is not set
+CONFIG_PWM_PCA9685=m
+CONFIG_IRQCHIP=y
 # CONFIG_IPACK_BUS is not set
 CONFIG_RESET_CONTROLLER=y
 # CONFIG_FMC is not set
@@ -3771,6 +3852,8 @@ CONFIG_CRC7=m
 # CONFIG_RANDOM32_SELFTEST is not set
 CONFIG_ZLIB_INFLATE=y
 CONFIG_ZLIB_DEFLATE=m
+CONFIG_LZO_COMPRESS=y
+CONFIG_LZO_DECOMPRESS=y
 # CONFIG_XZ_DEC is not set
 # CONFIG_XZ_DEC_BCJ is not set
 CONFIG_DECOMPRESS_GZIP=y
@@ -3786,6 +3869,7 @@ CONFIG_ARCH_HAS_ATOMIC64_DEC_IF_POSITIVE=y
 CONFIG_AVERAGE=y
 # CONFIG_CORDIC is not set
 # CONFIG_DDR is not set
+CONFIG_LIBFDT=y
 CONFIG_UCS2_STRING=y
 CONFIG_FONT_SUPPORT=y
 CONFIG_FONT_8x16=y
-- 
1.9.1

